src/backend/tcop/postgres.c

```diff
@@ -5314,6 +5336,24 @@ PostgresMain(int argc, char *argv[], const char *username)
 	return 1;					/* keep compiler quiet */
 }
 
+static void
+curl_finalize(int code, Datum arg __attribute__((unused)))
+{
+	if (curl_context.hasInited)
+	{
+		if (curl_context.response.buffer != NULL) {
+			pfree(curl_context.response.buffer);
+		}
+		/* cleanup curl stuff */
+		if (curl_context.curl_handle) {
+			curl_easy_cleanup(curl_context.curl_handle);
+		}
+		/* we're done with libcurl, so clean it up */
+		curl_global_cleanup();
+		curl_context.hasInited = false;
+		elog(LOG, "finalize the global struct for curl handle context.");
```

Comment:
I did not look into the where should be set as DEBUG*, but at least if the logs could be shown on our log files by default, they should 1) be accurate and helpful. 2) benefit not just developers 3) avoid potential flooding.